### PR TITLE
Implement basic login

### PR DESCRIPTION
**Pull request by llabake**
#### Sign In
- use the session method to maintain state 
- create a login form to create a new session to log a user in
- use flash.now to show flash messages on rendered page

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  def new
+  end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user && user.authenticate(params[:session][:password])
+      log_in user
+      redirect_to user
+    else
+      flash.now[:danger] = 'Invalid email/password combination' # Not quite right!
```
Comment (chalchuck) [inline app/controllers/sessions_controller.rb]: Why do you think it's not Right?

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
```
Comment (chalchuck) [inline app/controllers/sessions_controller.rb:5]: This method does not adhere to single responsibility principle.

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+module SessionsHelper
+  # Logs in the given user.
+  def log_in(user)
+    session[:user_id] = user.id
+  end
+
+  # Returns the current logged-in user (if any).
```
Comment (chalchuck) [inline app/helpers/sessions_helper.rb]: No comments please, let your code do the talking without the comments

**Review by chalchuck — COMMENTED** [review]
Please review and amend. 
I would really appreciate that at all times, when doing any piece of code, have the SOLID principles at back of your head.
And the code in the tutorial book is a just but a guide, read understand and then write the code.

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (llabake) [reply app/controllers/sessions_controller.rb]: 🤦‍♂️

**Pull request merged**